pkg/cmd/server/kubernetes/master/controller_volume.go

```diff
@@ -0,0 +1,158 @@
+package master
+
+import (
+	"fmt"
+
+	"github.com/openshift/origin/pkg/cmd/server/bootstrappolicy"
+
+	kv1core "k8s.io/client-go/kubernetes/typed/core/v1"
+	kclientv1 "k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/tools/record"
+	kctrlmgr "k8s.io/kubernetes/cmd/kube-controller-manager/app"
+	kubecontroller "k8s.io/kubernetes/cmd/kube-controller-manager/app"
+	kapi "k8s.io/kubernetes/pkg/api"
+	kapiv1 "k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/apis/componentconfig"
+	"k8s.io/kubernetes/pkg/cloudprovider"
+	attachdetachcontroller "k8s.io/kubernetes/pkg/controller/volume/attachdetach"
+	persistentvolumecontroller "k8s.io/kubernetes/pkg/controller/volume/persistentvolume"
+	"k8s.io/kubernetes/pkg/volume"
+	"k8s.io/kubernetes/pkg/volume/aws_ebs"
+	"k8s.io/kubernetes/pkg/volume/azure_dd"
+	"k8s.io/kubernetes/pkg/volume/cinder"
+	"k8s.io/kubernetes/pkg/volume/flexvolume"
+	"k8s.io/kubernetes/pkg/volume/gce_pd"
+	"k8s.io/kubernetes/pkg/volume/glusterfs"
+	"k8s.io/kubernetes/pkg/volume/host_path"
+	"k8s.io/kubernetes/pkg/volume/nfs"
+	"k8s.io/kubernetes/pkg/volume/rbd"
+	"k8s.io/kubernetes/pkg/volume/vsphere_volume"
+)
+
+type PersistentVolumeControllerConfig struct {
+	RecyclerImage string
+	CloudProvider cloudprovider.Interface
+}
+
+func (c *PersistentVolumeControllerConfig) RunController(ctx kubecontroller.ControllerContext) (bool, error) {
+	alphaProvisioner, err := kctrlmgr.NewAlphaVolumeProvisioner(c.CloudProvider, ctx.Options.VolumeConfiguration)
+	if err != nil {
+		return false, fmt.Errorf("A backward-compatible provisioner could not be created: %v, but one was expected. Provisioning will not work. This functionality is considered an early Alpha version.", err)
+	}
+
+	eventcast := record.NewBroadcaster()
+	recorder := eventcast.NewRecorder(kapi.Scheme, kclientv1.EventSource{Component: "persistent-volume-binder"})
```

Comment:
@deads2k this changed from "persistent-volume-controller", is that ok? (want to be consistent with SA name